config/devices/config.go

```diff
@@ -131,4 +131,42 @@ func Configure(p *config.Provider) {
 		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Devices", "IotHubs", "iothub_name",
 			"IotHubKeys", "name")
 	})
+
+	p.AddResourceConfigurator("azurerm_iothub_endpoint_storage_container", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"iothub_name": config.Reference{
+				Type: "IOTHub",
+			},
+			"container_name": config.Reference{
+				Type: rconfig.APISPackagePath + "/storage/v1alpha2.Container",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Devices/IotHubs/hub1/Endpoints/storage_container_endpoint1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Devices",
+			"IotHubs", "iothub_name",
+			"Endpoints", "name")
+	})
+
+	p.AddResourceConfigurator("azurerm_iothub_fallback_route", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"iothub_name": config.Reference{
+				Type: "IOTHub",
+			},
+			"endpoint_names": config.Reference{
+				Type: "IOTHubEndpointStorageContainer",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.IdentifierFromProvider
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Devices/IotHubs/hub1/FallbackRoute/default
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Devices",
+			"IotHubs", "iothub_name",
+			"FallbackRoute", "default")
```

Comment:
I would prefer having the exception (regarding the value of `FallbackRoute` being the constant `default`) be implemented here with something like:
```suggestion
		r.ExternalName.GetIDFn = func(ctx context.Context, externalName string, parameters map[string]interface{}, providerConfig map[string]interface{}) (string, error) {
			p, err := common.GetFullyQualifiedIDFn("Microsoft.Devices", "IotHubs", "iothub_name")(ctx, externalName, parameters, providerConfig)
			if err != nil {
				return "", err
			}
			return fmt.Sprintf("%s/FallbackRoute/default", p), nil
		}
```

Reply:
@ulucinar that's what I tried to do first, but GetFullyQualifiedIDFn returns a function, not (string, error). So getting a string out of it is a bit problematic in this context unless I am missing something obvious

Reply:
I guess I get it now! Let me test it properly